src/Nethermind/Nethermind.Specs/ChainSpecStyle/ChainSpecBasedSpecProvider.cs

```diff
@@ -205,55 +223,61 @@ private void FillReleaseSpec(ReleaseSpec releaseSpec, long releaseStartBlock, ul
 
-            releaseSpec.IsEip1153Enabled = (_chainSpec.Parameters.Eip1153TransitionTimestamp ?? ulong.MaxValue) <= releaseStartTimestamp;
-            releaseSpec.IsEip3651Enabled = (_chainSpec.Parameters.Eip3651TransitionTimestamp ?? ulong.MaxValue) <= releaseStartTimestamp;
-            releaseSpec.IsEip3855Enabled = (_chainSpec.Parameters.Eip3855TransitionTimestamp ?? ulong.MaxValue) <= releaseStartTimestamp;
-            releaseSpec.IsEip3860Enabled = (_chainSpec.Parameters.Eip3860TransitionTimestamp ?? ulong.MaxValue) <= releaseStartTimestamp;
+            return releaseSpec;
         }
 
         public void UpdateMergeTransitionInfo(long? blockNumber, UInt256? terminalTotalDifficulty = null)
         {
             if (blockNumber is not null)
+            {
                 MergeBlockNumber = (ForkActivation)blockNumber;
+            }
+
             if (terminalTotalDifficulty is not null)
+            {
                 TerminalTotalDifficulty = terminalTotalDifficulty;
+            }
         }
 
         public ForkActivation? MergeBlockNumber { get; private set; }
 
         public UInt256? TerminalTotalDifficulty { get; private set; }
 
-        public IReleaseSpec GenesisSpec => _transitions.Length == 0 ? null : _transitions[0].Release;
+        public IReleaseSpec GenesisSpec => _transitions.Length == 0 ? null : _transitions[0].Spec;
 
         public IReleaseSpec GetSpec(ForkActivation activation)
         {
+            // TODO: Is this actually needed? Can this be tricked with invalid activation check if someone would fake timestamp from the future?
             if (activation.Timestamp is not null)
             {
-                for (int i = 0; i < _transitions.Length; i++)
+                if (_timestampOnlyTransitions.TryGetSearchedItem(activation, CompareTransitionOnTimestamp, out (ForkActivation Activation, ReleaseSpec Spec) timestampTransition))
                 {
-                    if (_transitions[i].Activation.Timestamp is null)
-                        continue;
-                    ulong transitionTimestamp = _transitions[i].Activation.Timestamp.Value;
-                    if (transitionTimestamp < activation.Timestamp)
+                    if (timestampTransition.Activation.Timestamp < activation.Timestamp
+                        && timestampTransition.Activation.BlockNumber > activation.BlockNumber)
                     {
-                        if (_transitions[i].Activation.BlockNumber > activation.BlockNumber)
-                        {
-                            if (_logger.IsWarn) _logger.Warn("Chainspec file is misconfigured! Timestamp transition is configured to happen before the last block transition.");
-                        }
+                        if (_logger.IsWarn) _logger.Warn("Chainspec file is misconfigured! Timestamp transition is configured to happen before the last block transition.");
                     }
                 }
             }
```

Comment:
I would delete this code. This could trigger also when someone else's Chainspec is misconfigured.

Reply:
I get it debugging advantages. But we could hide it under `#IF DEBUG` or some flag.

Reply:
Managed to simplify it